Kudu.Services/Functions/FunctionController.cs

```diff
@@ -22,6 +25,100 @@ public FunctionController(IFunctionManager manager, ITraceFactory traceFactory)
+            using (tracer.Step($"FunctionsController.CreateOrUpdate({name})"))
+            {
+                try
+                {
+                    var functionEnvelope = await Request.Content.ReadAsAsync<FunctionEnvelope>();
+                    return Request.CreateResponse(HttpStatusCode.Created, await _manager.CreateOrUpdate(name, functionEnvelope));
+                }
+                catch (FileNotFoundException ex)
+                {
+                    tracer.TraceError(ex);
+                    return Request.CreateResponse(HttpStatusCode.NotFound, ex.Message);
+                }
+                catch (InvalidOperationException ex)
+                {
+                    tracer.TraceError(ex);
+                    return Request.CreateResponse(HttpStatusCode.Conflict, ex.Message);
+                }
+            }
+        }
+
+        [HttpGet]
+        public async Task<HttpResponseMessage> List()
+        {
+            var tracer = _traceFactory.GetTracer();
+            using (tracer.Step("FunctionsController.list()"))
+            {
+                return Request.CreateResponse(HttpStatusCode.OK, await _manager.ListFunctionsConfig());
+            }
+        }
+
+        [HttpGet]
+        public async Task<HttpResponseMessage> Get(string name)
+        {
+            var tracer = _traceFactory.GetTracer();
+            using (tracer.Step($"FunctionsController.Get({name})"))
+            {
+                return Request.CreateResponse(HttpStatusCode.OK, await _manager.GetFunctionConfig(name));
+            }
+        }
+
+        [HttpDelete]
+        public HttpResponseMessage Delete(string name)
+        {
+            var tracer = _traceFactory.GetTracer();
+            using (tracer.Step($"FunctionsController.Delete({name})"))
+            {
+                try
+                {
+                    _manager.DeleteFunction(name);
+                    return Request.CreateResponse(HttpStatusCode.NoContent);
+                }
+                catch (FileNotFoundException ex)
+                {
+                    tracer.TraceError(ex);
+                    return Request.CreateResponse(HttpStatusCode.NotFound, ex.Message);
+                }
+
+            }
+        }
+
+        [HttpGet]
+        public async Task<HttpResponseMessage> GetHostSettings()
+        {
+            var tracer = _traceFactory.GetTracer();
+            using (tracer.Step("FunctionsController.GetHostSettings()"))
+            {
+                return Request.CreateResponse(HttpStatusCode.OK, await _manager.GetHostConfig());
+            }
+        }
+
+        [HttpPut]
+        public async Task<HttpResponseMessage> PutHostSettings()
+        {
+            var tracer = _traceFactory.GetTracer();
+            using (tracer.Step("FunctionsController.PutHostSettings()"))
+            {
+                return Request.CreateResponse(HttpStatusCode.Created, await _manager.PutHostConfig(await Request.Content.ReadAsAsync<JObject>()));
```

Comment:
Shouldn't we only return 201 created if we created a new resource, distinguished from if we've updated an existing resource? Or simply return a 200?